src/index.tsx

```diff
@@ -53,8 +62,32 @@ checkConfigExists();
 
 function bootstrap() {
   const config = window.jujuDashboardConfig;
+  let error: string | null = null;
   if (!config) {
-    console.error("No configuration found.");
+    error = "No configuration found.";
+  }
+  const addressRegex = new RegExp(/^ws[s]?:\/\/(\S+)\/api$/);
+  if (!config?.controllerAPIEndpoint) {
+    error = `controllerAPIEndpoint is not set.`;
+  } else if (!config?.controllerAPIEndpoint.endsWith("/api")) {
+    error = `controllerAPIEndpoint (${config?.controllerAPIEndpoint}) must end with /api.`;
```

Comment:
Another possibility here is to postfix the API endpoint with `/api`,  but it's up to you whether you want it to be specifically set by the user or not.

Reply:
I did wonder about this, but the controllerAPIEndpoint needs to match values returned by Juju so I thought it was probably best (and less error prone) if this value is set correctly in the config (e.g. we could easily forget to append "/api" somewhere else in the code if the value is read straight from the config).